### ISSUE TITLE
A Tiny Mistake in Demo code of SWA (Stochastic Weight Averaging)

### ISSUE THREAD

**Issue by Zrealshadow** [pytorch/pytorch#46583]
## 📚 Documentation

url: https://pytorch.org/docs/stable/optim.html#stochastic-weight-averaging
In Stochastic_Weight_Averaging / Putting_it_all_together chapter

The displayed demo code:

```python
swa_model = torch.optim.swa_utils.AveragedModel(model)
scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=300)
swa_start = 160
swa_scheduler = SWALR(optimizer, swa_lr=0.05)

for epoch in range(300):
       for input, target in loader:
           optimizer.zero_grad()
           loss_fn(model(input), target).backward()
           optimizer.step()
       if i > swa_start:
           swa_model.update_parameters(model)
           swa_scheduler.step()
       else:
           scheduler.step()
 # Update bn statistics for the swa_model at the end
torch.optim.swa_utils.update_bn(loader, swa_model)
 # Use swa_model to make predictions on test data
preds = swa_model(test_input)
```

Mistake is In this line
```python
     if i > swa_start:
```
should be  this
```python
     if epoch > swa_start:
```

Just a Tiny mistake. QAQ

cc @jlin27 @mruberry @vincentqb

**Comment by zou3519** [comment]
Thanks for reporting @Zrealshadow. We would accept a patch that fixes this

**Comment by vincentqb** [comment]
Thanks for reporting! This was fixed in #45944. The doc online should be refreshed with the next sync.
